src/Profiles.ts

```diff
@@ -592,6 +553,44 @@ export class Profiles {
         return directProfile;
     }
 
+    private async deletePrompt(deletedProfile: IProfileLoaded) {
+        const profileName = deletedProfile.name;
+        this.log.debug(localize("deleteProfile.log.debug", "Deleting profile ") + profileName);
+        const quickPickOptions: vscode.QuickPickOptions = {
+            placeHolder: localize("deleteProfile.quickPickOption", "Are you sure you want to permanently delete ") + profileName,
+            ignoreFocusOut: true,
+            canPickMany: false
+        };
+        // confirm that the user really wants to delete
+        if (await vscode.window.showQuickPick([localize("deleteProfile.showQuickPick.yes", "Yes"),
+            localize("deleteProfile.showQuickPick.no", "No")], quickPickOptions) !== localize("deleteProfile.showQuickPick.yes", "Yes")) {
+            this.log.debug(localize("deleteProfile.showQuickPick.log.debug", "User picked no. Cancelling delete of profile"));
+            return;
+        }
+
+        try {
+            this.deleteProfileOnDisk(deletedProfile, profileName, deletedProfile.type);
```

Comment:
If the `deletedProfile` already hast the infromation about the type, why does the `deleteProfileOnDisk` function requires 3 parameters? (is there a `Closure` problem?)

Reply:
You are right. I missed that.

Reply:
No worries, I was just wondering if there was a problem accessing the value due to closure issues. So, it's fine 😋